app/models/miq_task.rb

```diff
@@ -259,42 +259,54 @@ def task_results=(value)
     binary_blob.binary = YAML.dump(value)
   end
 
-  def self.generic_action_with_callback(options, queue_options)
-    # Pre-reqs:
-    # options hash contains the following required keys:
-    #   :action => the human friendly name of the action to be run
-    #   :userid => the user this is being run for... aka, the logged on user who invoked the action in the UI
-    #
-    # queue options is a hash containing the following required keys:
-    #   :class_name
-    #   :method_name
-    #   :args
-    # queue_options keys that are not required but may be needed:
-    #   :instance_id (if using an instance method...an id)
-    #   :queue_name (which queue, priority?)
-    #   :zone (zone of the request)
-    #   :guid (guid of the server to run the action)
-    #   :role (role of the server to run the action)
-    #   :msg_timeout => how long you want to wait before pulling the plug on the action (seconds)
-
-    msg =  "Queued the action: [#{options[:action]}] being run for user: [#{options[:userid]}]"
-    task = MiqTask.create(
-      :name    => options[:action],
-      :userid  => options[:userid],
-      :state   => STATE_QUEUED,
-      :status  => STATUS_OK,
-      :message => msg)
+  # Create an MiqQueue object with an associated MiqTask object as its callback.
+  # Returns the ID of the generated task, or the full task object if the
+  # +return_task_object+ argument is set to true.
+  #
+  # Pre-reqs:
+  #
+  # The +options+ hash must contain the following required keys:
+  #
+  #   :name   => the human friendly name of the action to be run
+  #   :userid => the user this is being run for, i.e the logged on user who invoked the action in the UI
+  #
+  # You may alternatively specify :action instead of :name. All other options
+  # are passed through the task.
+  #
+  # The +queue_options+ is a hash containing the following required keys:
+  #
+  #   :class_name
+  #   :method_name
+  #   :args
+  #
+  # The +queue_options+ keys that are not required but may be needed:
+  #
+  #   :instance_id (if using an instance method...an id)
+  #   :queue_name (which queue, priority?)
+  #   :zone (zone of the request)
+  #   :guid (guid of the server to run the action)
+  #   :role (role of the server to run the action)
+  #   :msg_timeout => how long you want to wait before pulling the plug on the action (seconds)
+  #
+  def self.generic_action_with_callback(options, queue_options, return_task_object = false)
+    options[:name] ||= options.delete(:action) # Backwards compatibility
+    options = {:state => STATE_QUEUED, :status => STATUS_OK, :message => msg}.merge(options)
+
+    msg = "Queued the action: [#{options[:name]}] being run for user: [#{options[:userid]}]"
```

Comment:
This is being used before being defined, needs to be bumped up

Reply:
Oops, fixed.